dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -1257,4 +1239,37 @@ private void setRequestMetadata(Contentlet contentlet, HttpServletRequest reques
 			
 		}
 	}
+
+    private Map<Relationship, List<Contentlet>> evaluateContentRelationships(Contentlet contentlet, Structure st, Map<String,Object> map) throws DotDataException, DotSecurityException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 185 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)